pinot-core/src/main/java/org/apache/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -148,8 +153,11 @@ int getPriority(BaseFilterOperator filterOperator) {
         if (filterOperator instanceof OrFilterOperator) {
           return 3;
         }
+        if (filterOperator instanceof TextMatchFilterOperator) {
```

Comment:
why is it below AND/OR? 
lets say the query is  (a>10 or b<10) and (text_match(c, "<some text>"). It's better to evaluate text match before evaluating OR and use the results from text_match to restrict the scope of OR.
I think the priority of this should be similar to bitmap. WDYT?

Reply:
It is going to happen like that only since TextMatch will return BitMapDocIdSet and the AndBlockDocIdSet is intelligent about keeping bitmaps at the top and append the ones in remaining category (child AND/OR) and the scan based category (at the end. 

I have adjusted the priority to 2. So now the order is

0 - Sort based inverted index
1 - bitmap inverted index
2 - text match
3 - AND
4 - OR
5 - SCAN
10 - Regex

Since 1 and 2 both return bitmaps, AndBlockDocIdSet will intersect them upfront before creating the AndDocIdIterator -- this will take care of multiple text_match() predicates or text_match() along with a predicate using native Pinot inverted index.